solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -210,6 +212,10 @@ public void addCloseHook(Closeable hook) {
     }
   }
 
+  public QueryLimits getLimits() {
+    return (QueryLimits) req.getContext().computeIfAbsent(LIMITS_KEY, (k) -> new QueryLimits(req));
```

Comment:
This may throw NPE if used in a wrong place - HttpSolrCall.sendRemoteQuery() uses the other constructor for SolrRequestInfo, where req is null.

Reply:
Yeah I need to understand better how req is populated.